presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -144,6 +145,45 @@ public PlanWithConsumedDynamicFilters visitJoin(JoinNode node, Set<String> allow
             return new PlanWithConsumedDynamicFilters(node, ImmutableSet.copyOf(consumed));
         }
 
+        @Override
+        public PlanWithConsumedDynamicFilters visitSemiJoin(SemiJoinNode node, Set<String> allowedDynamicFilterIds)
+        {
+            ImmutableSet<String> allowedDynamicFilterIdsProbeSide = ImmutableSet.<String>builder()
+                    .addAll(node.getDynamicFilters().keySet())
+                    .addAll(allowedDynamicFilterIds)
+                    .build();
+
+            PlanWithConsumedDynamicFilters leftResult = node.getSource().accept(this, allowedDynamicFilterIdsProbeSide);
+            Set<String> consumedProbeSide = leftResult.getConsumedDynamicFilterIds();
+            Map<String, VariableReferenceExpression> dynamicFilters = node.getDynamicFilters().entrySet().stream()
+                    .filter(entry -> consumedProbeSide.contains(entry.getKey()))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            PlanWithConsumedDynamicFilters rightResult = node.getFilteringSource().accept(this, allowedDynamicFilterIds);
+            Set<String> consumed = new HashSet<>(rightResult.getConsumedDynamicFilterIds());
+            consumed.addAll(consumedProbeSide);
+            consumed.removeAll(dynamicFilters.keySet());
```

Comment:
This can be merged with `visitJoin`'s common part just like how you did for the impl part.